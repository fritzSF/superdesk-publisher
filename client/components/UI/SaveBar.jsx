import React from "react";
import PropTypes from "prop-types";
import { Button } from "superdesk-ui-framework/react";

const SaveBar = ({ save, cancel, isDisabled }) => {
  return (
    <div className="sd-collapse-box__sliding-toolbar-wrapper">
      <div className="sd-collapse-box__sliding-toolbar">
        <div className="sliding-toolbar__inner" />
        <Button text="Cancel" style="hollow" theme="dark" onClick={cancel} />
        <Button
          text="Save"
          type="primary"
          theme="dark"
          onClick={save}
          disabled={isDisabled}
          data-testid="save-button"
        />
      </div>
    </div>
  );
};

SaveBar.propTypes = {
  save: PropTypes.func.isRequired,
  cancel: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool,
};

SaveBar.defaultProps = {
  isDisabled: false,
};

export default SaveBar;
