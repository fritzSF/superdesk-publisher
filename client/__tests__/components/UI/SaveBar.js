import React from "react";
import SaveBar from "../../../components/UI/SaveBar";
import { render, fireEvent } from "@testing-library/react";

describe("UI/SaveBar", () => {
  it("renders correctly", () => {
    const { container } = render(
      <SaveBar save={jest.fn()} cancel={jest.fn()} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("save function fired", () => {
    const save = jest.fn();
    const { getByText } = render(<SaveBar save={save} cancel={jest.fn()} />);
    const button = getByText(/save/i);

    fireEvent.click(button);
    expect(save).toHaveBeenCalled();
  });

  it("cancel function fired", () => {
    const cancel = jest.fn();
    const { getByText } = render(<SaveBar cancel={cancel} save={jest.fn()} />);

    fireEvent.click(getByText(/cancel/i));
    expect(cancel).toHaveBeenCalled();
  });

  it("isDisabled blocks save button", () => {
    const save = jest.fn();
    const { container } = render(
      <SaveBar save={save} isDisabled={true} cancel={jest.fn()} />
    );
    const button = container.querySelector("button");

    fireEvent.click(button);
    expect(save).not.toHaveBeenCalled();
  });
});
