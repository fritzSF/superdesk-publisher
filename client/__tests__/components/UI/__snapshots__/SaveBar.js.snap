// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/SaveBar renders correctly 1`] = `
<div
  class="sd-collapse-box__sliding-toolbar-wrapper"
>
  <div
    class="sd-collapse-box__sliding-toolbar"
  >
    <div
      class="sliding-toolbar__inner"
    />
    <button
      aria-label=""
      class="btn btn--hollow btn--ui-dark"
    >
      Cancel
    </button>
    <button
      aria-label=""
      class="btn btn--primary btn--ui-dark"
    >
      Save
    </button>
  </div>
</div>
`;
